extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java

```diff
@@ -146,6 +146,12 @@ private static void prepareBouncyCastleProvider(BuildProducer<ReflectiveClassBui
         reflection.produce(new ReflectiveClassBuildItem(true, true,
                 isFipsMode ? SecurityProviderUtils.BOUNCYCASTLE_FIPS_PROVIDER_CLASS_NAME
                         : SecurityProviderUtils.BOUNCYCASTLE_PROVIDER_CLASS_NAME));
+        reflection.produce(new ReflectiveClassBuildItem(true, true,
+                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyPairGeneratorSpi"));
+        reflection.produce(new ReflectiveClassBuildItem(true, true,
+                "org.bouncycastle.jcajce.provider.asymmetric.ec.KeyPairGeneratorSpi$EC"));
+        reflection.produce(new ReflectiveClassBuildItem(true, true,
+                "org.bouncycastle.jcajce.provider.asymmetric.rsa.KeyPairGeneratorSpi"));
```

Comment:
@stuartwdouglas, would it make sense to use the recently added weak reflection support[1] here? I see that the `ReflectiveClassBuildItem` by default isn't a "weak" one.

[1] https://github.com/quarkusio/quarkus/pull/19115

Reply:
@jaikiran Not sure it will optimize anything in this case, `KeyPairGeneratorSpi` is no accessed directly. Perhaps it can make sense to open a separate issue to review BC related classes which are registered for the registration as there are a few of them already regstered so it makes sense to check all of them - please do it if you'd like, as I would prefer to have this PR fixing a linked issue such that it can also be backported back to `2.1`.

Reply:
Hello @sberyozkin, no objections to merging this PR in the current form :) I was just curious whether the weak reflection will help in cases like these where not all of these classes might get used at runtime.

Reply:
@jaikiran Hi, sure, I was actually looking at `ReflectiveClassBuildItem` before opening a PR and wondering what a `weak` flag was about, so thanks for asking Stuart about possibly applying it in this case. But indeed, since it is a new feature, I'd rather look at applying, if it is applicable, in a separate dedicated PR.
My understanding that classes like `org.bouncycastle.jcajce.provider.asymmetric.rsa.KeyPairGeneratorSpi` would not be considered directly reachable since they are only registered indirectly by the BC provider at the static/or runtime initialization time and are accessed indirectly via a standard Java Security API like `KeyPairGenerator.getInstance("RSA", "BC")` - but it is very likely I'm not seeing it correctly - so indeed it can make sense to look into it further